src/controllers/cart.js

```diff
@@ -41,29 +41,27 @@ exports.postCartItemByDishId = async (req, res, next) => {
 
     let cartItem = await Cart.findOne({
       customerId: req.user.id,
-      'cartItems.dishId': req.params.id,
+      'cartItems.dishId': req.body.id,
     });
     if (cartItem) return next(new CustomError('item already in cart', 300));
 
     cartItem = {
-      dishId: req.params.id,
+      dishId: req.body.id,
       quantity: req.body.quantity ? req.body.quantity : 1,
     };
     cart.cartItems.push(cartItem);
     cart.save();
     return res.status(201).json(cart);
   } catch (err) {
-    return res
-      .status(500)
-      .json({ message: 'Failed to add item to cart', error: err });
+    return next(new CustomError('Failed to add item to cart'));
```

Comment:
same thing here instead of 'Failed to delete cart' use err.message and 500 status code